client/src/node.rs

```diff
@@ -298,6 +300,42 @@ impl Node {
         self.config.chain.chain_id
     }
 
+    pub fn syncing(&self) -> Result<SyncingStatus> {
+        let head_age = self.check_head_age();
+        let syncing = match head_age {
+            Ok(_) => false,
+            Err(_) => true,
+        };
+
+        if !syncing {
+            Ok(SyncingStatus::IsFalse)
+        } else {
+            let latest_synced_block = self.get_block_number().unwrap();
+            let oldest_payload = self.payloads.first_key_value();
+            let oldest_synced_block = *oldest_payload.ok_or(eyre!("First block not found"))?.0;
+            let highest_block = self.consensus.expected_current_slot();
+            Ok(SyncingStatus::IsSyncing(Box::new(SyncProgress {
+                current_block: (latest_synced_block.into()),
+                highest_block: (highest_block.into()), 
+                starting_block: (oldest_synced_block.into()),
+                pulled_states: None,
```

Comment:
Default doesnt exist on SyncProgress unfortunately.


pub struct SyncProgress {
    pub current_block: U64,
    pub highest_block: U64,
    pub starting_block: U64,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub pulled_states: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub known_states: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healed_bytecode_bytes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healed_bytecodes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healed_trienode_bytes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healed_trienodes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healing_bytecode: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub healing_trienodes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_account_bytes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_accounts: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_bytecode_bytes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_bytecodes: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_storage: Option<U64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub synced_storage_bytes: Option<U64>,
}

Any other suggestions to avoid initializing each as None?

Reply:
Usually it is just the members that need to have default values. Options default to none. Does this work?

```rust
SyncProgress {
    current_block: latest_synced_block.into(),
    highest_block: highest_block.into(),
    starting_block: oldest_synced_block.into(),
    ..default::Default()
}
```

Reply:
I tried this. I get the following...

..Default::default()
    |                   ^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `SyncProgress`
    
 Basically SyncProgress uses serde and doesn't correctly initialize a default struct, so we can't really use ..Default::default() without creating an actual default here of None.

Reply:
Oh ok that makes sense. In that case, once you just fix up the one error handling comment I left in rpc, I will merge.